barretenberg/cpp/src/barretenberg/vm/avm/trace/execution.cpp

```diff
@@ -359,12 +356,15 @@ std::vector<Row> Execution::gen_trace(AvmPublicInputs const& public_inputs,
 
         vinfo("Beginning execution of phase ", to_name(phase), " (", public_call_requests.size(), " enqueued calls).");
         AvmError phase_error = AvmError::NO_ERROR;
-        for (auto public_call_request : public_call_requests) {
+        for (size_t i = 0; i < public_call_requests.size(); i++) {
+            auto public_call_request = public_call_requests.at(i);
```

Comment:
hmm i think i had to go back to this index-looping because i need to grab the `enqueued_call_hint` for the same `public_call_request` (3 lines below)
`auto enqueued_call_hint = execution_hints.enqueued_call_hints.at(i);`